File: platforms/ios/Dezel/Dezel/View/WebViewObserver.swift
/**
 * @protocol WebViewObserver
 * @since 0.7.0
 */
public protocol WebViewObserver : AnyObject {
	func willLoad(webView: WebView, url: URL) -> Bool
	func didLoad(webView: WebView)
	func didUpdateContentSize(webView: WebView, size: CGSize)
}

File: platforms/android/Dezel/dezel/src/main/java/ca/logaritm/dezel/view/WebViewObserver.kt
package ca.logaritm.dezel.view

import android.util.Size

/**
 * @protocol WebViewObserver
 * @since 0.7.0
 */
public interface WebViewObserver {

	/**
	 * @method onBeforeLoad
	 * @since 0.7.0
	 */
	fun onBeforeLoad(webView: WebView, url: String): Boolean

	/**
	 * @method onLoad
	 * @since 0.7.0
	 */
	fun onLoad(webView: WebView)

	/**
	 * @method onUpdateContentSize
	 * @since 0.7.0
	 */
	fun onUpdateContentSize(webView: WebView, size: Size)
}